lib/import.js

```diff
@@ -95,21 +128,68 @@ module.exports = (self) => {
       }
     },
 
-    async importPieces (req, pieces, reporting) {
-      let importedCount = 0;
+    async importOrUpdatePieces (req, {
+      pieces, reporting, updateKey, updateField
+    }) {
+      const counter = {
+        imported: 0,
+        updated: 0,
+        failed: 0
+      };
 
       for (const piece of pieces) {
         try {
-          await self.insert(req, piece);
+          const updateKeyValue = updateKey && piece[updateKey];
+
+          if (!updateKeyValue) {
+            const converted = {};
+
+            await self.convert(req, piece, converted);
+            await self.insert(req, converted);
 
+            counter.imported++;
+            reporting.success();
+            continue;
+          }
+
+          const existingPiece = await self.findForEditing(req, {
+            aposMode: 'draft',
+            [updateField]: updateKeyValue
+          }).toObject();
+
+          if (!existingPiece) {
+            throw new Error('No piece to update');
+          }
+
+          if (!piece[updateField]) {
+            piece[updateField] = updateKeyValue;
+          }
```

Comment:
I had to do that because, in the case of an update, when a required field is used as key, the convert method was throwing an error saying that the `field` is required, but this field already exist in the existing piece. So the convert only look in the input document not the converted one, Weirdly I tested in Apostrophe 2 and it was working.
Of course of the field is present twice, once as a key, once as a normal field, it will take the normal field.